mitodl/micromasters
# Attach user to coupon

#### What are the relevant tickets?
Fixes #2323.

#### What's this PR do?
When the user visits the dashboard with a `coupon` query parameter in the URL, that query parameter is assumed to be a coupon code. This PR detects that query parameter, and tries to attach the currently logged-in user to that coupon. If the coupon is invalid or doesn't exist, we pop a toast notification that it didn't work. If the coupon exists and is valid for the user, we pop a notification to confirm that they have successfully applied the coupon.

#### How should this be manually tested?
* Try applying a coupon code for a coupon that doesn't exist, and confirm that you get a notification.
* Then make a coupon that is invalid, and confirm you get the same notification when you try to apply it.
* Make a percentage discount coupon that can be successfully applied, and apply it. You should get a confirmation notification, with information about the coupon you just applied.
* Try again with a fixed discount coupon that can be successfully applied. Make sure that the information in the confirmation notification is accurate.

#### Screenshots (if appropriate)

![fixed discount notification](https://cloud.githubusercontent.com/assets/132355/22153480/7da7c0e4-def5-11e6-88b4-c8b87a196ec4.png)
![percent discount notification](https://cloud.githubusercontent.com/assets/132355/22153481/7dad7c50-def5-11e6-9333-70e799ed9313.png)
![coupon failure notification](https://cloud.githubusercontent.com/assets/132355/22153479/7da7396c-def5-11e6-83b8-86a1d7ff53de.png)
